src/software/thunderscope/robot_diagnostics/drive_and_dribbler_widget.py

```diff
@@ -7,6 +8,93 @@
 from software.thunderscope.proto_unix_io import ProtoUnixIO
 from software.thunderscope.common import common_widgets
 
+import evdev
+from evdev import ecodes
+
+
+class KeyboardMagic:
+    def __init__(self, drive_and_dribble_widget):
+        self.drive_and_dribble_widget = drive_and_dribble_widget
+        self.device = evdev.InputDevice("/dev/input/event3")
+        self.thread = Thread(target=self.read_one_loop)
+        self.thread.start()
+
+        self.motor_control = MotorControl()
+        self.motor_control.direct_velocity_control.velocity.x_component_meters = 0
+        self.motor_control.direct_velocity_control.velocity.y_component_meters = 0
+
+    def read_one_loop(self):
+        for event in self.device.read_loop():
+            # motor_control.direct_velocity_control.velocity.x_component_meters = (
+            #    -move_axis_x * self.constants.robot_max_speed_m_per_s * speed_factor
+            # )
+
+            # motor_control.direct_velocity_control.velocity.y_component_meters = (
+            #    -move_axis_y * self.constants.robot_max_speed_m_per_s * speed_factor
+            # )
+
+            if event.code == ecodes.KEY_S and event.value == 1:
+                self.motor_control.direct_velocity_control.velocity.x_component_meters = (
```

Comment:
probably can use some python magic if you want:
`self.motor_control.direct_velocity_control.velocity.x_component_meters = 3.0/2.0 if event.value == 1 else 0`